tpl/math/math_test.go

```diff
@@ -153,6 +153,51 @@ func TestLog(t *testing.T) {
 		c.Assert(err, qt.IsNil)
 		c.Assert(result, qt.Equals, test.expect)
 	}
+
+	// Separate test for Log(-1) -- returns NaN
+	result, err := ns.Log(-1)
+	c.Assert(err, qt.IsNil)
+	c.Assert(result, qt.Satisfies, math.IsNaN)
+}
+
+func TestSqrt(t *testing.T) {
+	t.Parallel()
+	c := qt.New(t)
+
+	ns := New()
+
+	for _, test := range []struct {
+		a      interface{}
+		expect interface{}
+	}{
+		{81, float64(9)},
+		{0.25, float64(0.5)},
+		{0, float64(0)},
+		{"abc", false},
+	} {
```

Comment:
Add a test for negative values.

Reply:
I meant to edit this comment before submitting it after seeing your test for negative at the end of the func.  Can we incorporate that test into this table-based test loop?

Reply:
I didn't include the negativity test in the existing loop because it required using `qt.Satisfies` instead of `qt.Equals`, and I couldn't see a neater way (not being familiar with QuickTest).  Any suggestions for a better way?

Reply:
If you take my advice on changing the Sqrt error handling, will you get a NaN result?  I suggested returning zero, but I suppose you could return NaN if `af < 0`.  If you do return NaN, you can make a special case in the loop similar to the `test.expect.(bool)` case on line 181 below.

I'm probably being a little pedantic here.  I just prefer having all the tests in the table if we're going to use table-based tests.

Reply:
I don't think `math.Sqrt(-1)` should return zero because that's wrong maths, and wouldn't alert the user to the problem.  

An alternative would be to raise an error, as it currently does if the argument is not numeric.  But Hugo is quite capable of handling NaN values -- for example `printf "%f" (math.Sqrt -1)` displays the string "NaN" which would make the problem visible without crashing anything.

So I still think that returning NaN is the best approach.  

And while I agree that keeping the test suite table-based would be ideal, it's not possible to use `qt.Equals` with NaNs, so a separate test is needed.

Reply:
I'm going to decide here: We don't need to do anthing special here: Just return the value returned from `math.Sqrt`. Also, I'm not too interested in tests that tests math.Sqrt -- just smoke test it (verify that we call the right function) and then test some of the error cases (on the input side). Which makes this PR look perfectly fine to me as it is now.

Reply:
OK, thanks.